fdbserver/BlobManager.actor.cpp

```diff
@@ -1278,32 +1282,49 @@ ACTOR Future<Void> reevaluateInitialSplit(Reference<BlobManagerData> bmData,
 	// calculate new split targets speculatively assuming split is too large and current worker still owns it
 	ASSERT(granuleRange.begin < proposedSplitKey);
 	ASSERT(proposedSplitKey < granuleRange.end);
-	state Future<BlobGranuleSplitPoints> fSplitFirst = splitRange(bmData, KeyRangeRef(granuleRange.begin, proposedSplitKey), false, true);
-	state Future<BlobGranuleSplitPoints> fSplitSecond = splitRange(bmData, KeyRangeRef(proposedSplitKey, granuleRange.end), false, true);
+	state Future<BlobGranuleSplitPoints> fSplitFirst =
+	    splitRange(bmData, KeyRangeRef(granuleRange.begin, proposedSplitKey), false, true);
+	state Future<BlobGranuleSplitPoints> fSplitSecond =
+	    splitRange(bmData, KeyRangeRef(proposedSplitKey, granuleRange.end), false, true);
 
 	state Standalone<VectorRef<KeyRef>> newRanges;
 
 	BlobGranuleSplitPoints splitFirst = wait(fSplitFirst);
-	ASSERT(splitFirst.size() >= 2);
-	ASSERT(splitFirst.front() == granuleRange.begin);
-	ASSERT(splitFirst.back() == proposedSplitKey);
+	ASSERT(splitFirst.keys.size() >= 2);
+	ASSERT(splitFirst.keys.front() == granuleRange.begin);
+	ASSERT(splitFirst.keys.back() == proposedSplitKey);
 	for (int i = 0; i < splitFirst.keys.size(); i++) {
-		// FIXME: need to align proposedSplitKey once that's merged
 		newRanges.push_back_deep(newRanges.arena(), splitFirst.keys[i]);
 	}
 
 	BlobGranuleSplitPoints splitSecond = wait(fSplitSecond);
-	ASSERT(splitSecond.size() >= 2);
-	ASSERT(splitSecond.front() == proposedSplitKey);
-	ASSERT(splitSecond.back() == granuleRange.end);
+	ASSERT(splitSecond.keys.size() >= 2);
+	ASSERT(splitSecond.keys.front() == proposedSplitKey);
+	ASSERT(splitSecond.keys.back() == granuleRange.end);
 	// i=1 to skip proposedSplitKey, since above already added it
 	for (int i = 1; i < splitSecond.keys.size(); i++) {
 		newRanges.push_back_deep(newRanges.arena(), splitSecond.keys[i]);
 	}
 
+	if (BM_DEBUG) {
+		fmt::print("Re-evaluated split ({0}:\n", newRanges.size());
+		for (auto& it : newRanges) {
+			fmt::print("    {0}\n", it.printable());
+		}
+	}
+
 	// redo key alignment on full set of split points
```

Comment:
I think we can align [begin, splitpoint, end) upfront and then we won't have to realign later.